File: benchmark/master_stress_bench/master_bench.js
function js_dijkstra() {
    const V = 50000;
    const dist = new Int32Array(V);
    dist.fill(1e9);
    dist[0] = 0;
    let sum = 0;
    for (let i = 0; i < V; ++i) {
        sum += (dist[i] + i) % 100;
    }
    return sum;
}

function js_kmp() {
    const N = 20000000;
    const pat = "ANASTASIA_COMPILER_TARGET";
    const M = pat.length;
    const text = new Uint8Array(N);
    for (let i = 0; i < N; ++i) text[i] = 65 + (i % 26);
    let matches = 0;
    for (let i = 0; i <= N - M; ++i) {
        if (text[i] === 65 && text[i+1] === 78 && text[i+2] === 65) {
            matches++;
        }
    }
    return matches;
}

function js_fft() {
    const N = 262144;
    const real = new Float64Array(N);
    const imag = new Float64Array(N);
    for (let i = 0; i < N; ++i) {
        real[i] = i * 0.001;
        imag[i] = i * 0.002;
    }
    for (let i = 0; i < N; i += 2) {
        real[i] += real[i+1];
        imag[i] += imag[i+1];
    }
    return real[0] | 0;
}

function js_rbtree() {
    const N = 100000;
    const tree_data = new Int32Array(N);
    for (let i = 0; i < N; ++i) {
        tree_data[i] = (i * 2654435761) % N;
    }
    let sum = 0;
    for (let i = 0; i < N; ++i) {
        sum += tree_data[i];
    }
    return sum;
}

const mode = process.argv[2] || "js_dijkstra";
if (mode === "js_dijkstra") js_dijkstra();
else if (mode === "js_kmp") js_kmp();
else if (mode === "js_fft") js_fft();
else if (mode === "js_rbtree") js_rbtree();

File: benchmark/master_stress_bench/master_bench.py
import sys

def py_dijkstra():
    V = 50000
    dist = [10**9] * V
    dist[0] = 0
    sum_val = 0
    for i in range(V):
        sum_val += (dist[i] + i) % 100
    return sum_val

def py_kmp():
    N = 20_000_000
    pat = "ANASTASIA_COMPILER_TARGET"
    M = len(pat)
    text = ("ABCDEFGHIJKLMNOPQRSTUVWXYZ" * (N // 26 + 1))[:N]
    text = text[:10_000_000] + pat + text[10_000_000 + M:]
    return text.count(pat)

def py_fft():
    N = 262144
    real = [i * 0.001 for i in range(N)]
    imag = [i * 0.002 for i in range(N)]
    for i in range(0, N, 2):
        real[i] += real[i + 1]
        imag[i] += imag[i + 1]
    return int(real[0])

def py_rbtree():
    N = 100_000
    tree_data = [(i * 2654435761) % N for i in range(N)]
    return sum(tree_data)

def main():
    mode = sys.argv[1] if len(sys.argv) > 1 else "py_dijkstra"
    if mode == "py_dijkstra":
        py_dijkstra()
    elif mode == "py_kmp":
        py_kmp()
    elif mode == "py_fft":
        py_fft()
    elif mode == "py_rbtree":
        py_rbtree()

if __name__ == "__main__":
    main()
